lib/oli_web/live_session_plugs/set_sidebar.ex

```diff
@@ -52,18 +52,23 @@ defmodule OliWeb.LiveSessionPlugs.SetSidebar do
 
     notes_enabled = collab_space_pages_count > 0
 
-    %{slug: revision_slug} = DeliveryResolver.root_container(section_slug)
+    revision_slug =
+      case DeliveryResolver.root_container(section_slug) do
+        %{slug: slug} -> slug
+        _ -> nil
+      end
 
     discussions_enabled =
-      case Collaboration.get_collab_space_config_for_page_in_section(
-             revision_slug,
-             section_slug
-           ) do
-        {:ok, %CollabSpaceConfig{status: :enabled}} ->
-          true
-
-        _ ->
-          false
+      if revision_slug do
+        case Collaboration.get_collab_space_config_for_page_in_section(
+               revision_slug,
+               section_slug
+             ) do
+          {:ok, %CollabSpaceConfig{status: :enabled}} -> true
+          _ -> false
+        end
+      else
+        false
       end
```

Comment:
This is an excellent opportunity to use a `with` statement:

```
    with %{slug: revision_slug} <- DeliveryResolver.root_container(section_slug),
         {:ok, %CollabSpaceConfig{status: :enabled}} <-
           Collaboration.get_collab_space_config_for_page_in_section(revision_slug, section_slug) do
      true
    else
      _ -> false
    end
```

When there is a well-defined happy path, `with` is a good building block to use.

Note: I haven't tested the suggested block; use it as a reference.